scripts/compress_images_test.py

```diff
@@ -0,0 +1,238 @@
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS-IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Tests for compress_images.py."""
+
+from __future__ import annotations
+
+import logging
+import os
+import pathlib
+import shutil
+import subprocess
+import tempfile
+from unittest import mock
+from core.tests import test_utils
+from scripts import compress_images
+
+from PIL import Image
+from typing import List, TypedDict
+
+
+class CompressedImageInfo(TypedDict):
+    """Type definition for compressed image information."""
+
+    path: pathlib.Path
+    original_size: int
+    new_size: int
+
+
+class TestImageCompressor(test_utils.GenericTestBase):
+    """Test the ImageCompressor class."""
+
+    def setUp(self) -> None:
+        """Set up test environment before each test."""
+        self.temp_dir = tempfile.mkdtemp()
+        self.output_dir = os.path.join(self.temp_dir, 'compressed_images')
+
+        self.png_path = os.path.join(self.temp_dir, 'test_image.png')
+        self.jpg_path = os.path.join(self.temp_dir, 'test_image.jpg')
+        self.webp_path = os.path.join(self.temp_dir, 'test_image.webp')
+
+        self._create_test_image(self.png_path, 'PNG')
+        self._create_test_image(self.jpg_path, 'JPEG')
+        self._create_test_image(self.webp_path, 'WEBP')
+
+    def tearDown(self) -> None:
+        """Clean up test environment after each test."""
+        shutil.rmtree(self.temp_dir)
+
+    def _create_test_image(self, path: str, Img_format: str) -> None:
+        """Create a test image for compression testing."""
+        img = Image.new('RGB', (100, 100), color='red')
+        img.save(path, format=Img_format)
+
+    def test_get_compression_type_zip_extension(self) -> None:
+        """Test get_compression_type returns 'Zip' for PNG."""
+        compressor = compress_images.ImageCompressor(self.temp_dir)
+        compression_type = compressor.get_compression_type('.png')
+        self.assertEqual(compression_type, 'Zip')
+
+    def test_get_compression_type_lzw_extension(self) -> None:
+        """Test get_compression_type returns 'LZW' for JPG."""
+        compressor = compress_images.ImageCompressor(self.temp_dir)
+        compression_type = compressor.get_compression_type('.jpg')
+        self.assertEqual(compression_type, 'LZW')
+
+    def test_compress_images(self) -> None:
+        """Test image compression process."""
+        with self.swap(subprocess, 'run', mock.MagicMock()):
+            self.assertFalse(os.path.exists(self.output_dir))
+            result_images: List[CompressedImageInfo] = [
+                {
+                    'path': pathlib.Path(self.png_path),
+                    'original_size': 1000,
+                    'new_size': 500
```

Comment:
this is odd because at the end of the test we assert that the image size hasn't changed. (it's also odd that the file sizes are the same since a single-color image should be easy to compress)